--- staging/runtime/gems/rubysl-fileutils-2.0.2/lib/rubysl/fileutils/fileutils.rb.orig	Tue Oct 22 09:04:50 2013
+++ staging/runtime/gems/rubysl-fileutils-2.0.2/lib/rubysl/fileutils/fileutils.rb	Tue Oct 22 09:05:25 2013
@@ -205,7 +205,7 @@
     list.map {|path| path.chomp(?/) }.each do |path|
       # optimize for the most common case
       begin
-        fu_mkdir path, options[:mode]
+        fu_mkdir path, options[:mode] unless File.directory?(path)
         next
       rescue SystemCallError
         next if File.directory?(path)
@@ -218,7 +218,7 @@
       end
       stack.reverse_each do |dir|
         begin
-          fu_mkdir dir, options[:mode]
+          fu_mkdir dir, options[:mode] unless File.directory?(path)
         rescue SystemCallError
           raise unless File.directory?(dir)
         end
