--- acroread.orig	Tue May 23 03:33:35 2006
+++ acroread	Sun Oct  8 20:57:19 2006
@@ -233,7 +233,7 @@
  ScriptDirectory="`/bin/pwd 2> /dev/null`"
  cd "$cwd"
 fi
-install_dir=`dirname "$ScriptDirectory"`/Reader
+install_dir=%%TRUEPREFIX%%/Acrobat7/Reader
 
 
 #
@@ -476,7 +476,7 @@
     OLD_IFS=""
 
     # This checks for libs in cache /etc/ld.so.cache
-    for ifile in `/sbin/ldconfig -p|grep "libgtk-x11-2.0.so$"|sed 's/.*[ \t]*=>[ \t]*\(.*\)/\1/g' 2>/dev/null`; do
+    for ifile in `/emul/linux/sbin/ldconfig -p|grep "libgtk-x11-2.0.so$"|sed 's/.*[ \t]*=>[ \t]*\(.*\)/\1/g' 2>/dev/null`; do
         version=`get_gtk_file_ver "$ifile"`
 
     	while [ ${#version} -gt ${#base_version} ]; do
@@ -610,7 +610,7 @@
         ;;
     esac
     ;;
-  Linux)
+  OpenBSD|Linux)
     ACRO_CONFIG=intellinux
     export ACRO_CONFIG
     ;;
